core/lib/spree/core/controller_helpers/auth.rb

```diff
@@ -57,10 +57,6 @@ def set_guest_token
           end
         end
 
-        def store_location
```

Comment:
Did you get a chance to see where this method was called from?

Reply:
As far as I could tell, it was only used in spec files which made it ripe for removal. Is this another public method in use?

Reply:
Oh, it is used by our authentication plugin (using devise) [here](https://github.com/solidusio/solidus_auth_devise/search?q=store_location). I think this should be taken into account to understand if we can remove the method and we are using the right approach.

Reply:
Agreed. Personally I would opt to just replace our usage of `redirect_back_or_default` with Rails's implementation, but indicate that the class that maintains that behavior has been deprecated. I'll ensure deprecation warnings are in place and put the specs back.

Reply:
I'm just concerned that we are deprecating a core function without providing an alternative for extensions using this method. This could even be just instructions on how to replace `store_location` with something else.

Reply:
Hey Andre, I just found [this page on the Devis Wiki](https://github.com/heartcombo/devise/wiki/How-To:-Redirect-back-to-current-page-after-sign-in,-sign-out,-sign-up,-update), which seems very relevant. I suspect we needed the custom code when the Devise gem did not have those methods available. Apparently we just can't rely on the header referral here, so some sort of storage mechanism for the last visited URL (except the authentication ones to avoid redirect loops) is somehow required.